sklearn/compose/_column_transformer.py

```diff
@@ -294,7 +307,12 @@ def _update_fitted_transformers(self, transformers):
         transformers = iter(transformers)
         transformers_ = []
 
-        for name, old, column in self.transformers:
+        transformer_iter = chain(self.transformers, [self._remainder])
+
+        for name, old, column in transformer_iter:
+            if column is None:
```

Comment:
why?

Reply:
If you are speaking of `transformer_iter`: It is used to iterate over the same number of items that `_iter. In `_iter`:

```python
transformers = chain(self.transformers, [self._remainder])
```

If you are speaking of `if column is None`: `_iter` also skips the transformer when column is None.

Reply:
> If you are speaking of `if column is None`: `_iter` also skips the transformer when column is None.

Where do you see this logic?

Reply:
I updated `_iter` to include this:

```python
for name, trans, column in transformers:
    if column is None:
        continue
```

This condition happens when there are no `self._remainder=(_, _, None)`, which occurs when `passthrough==None` or when there are no remaining columns left.

Reply:
well, maybe it's best to check that `name == 'remainder'` too. It would make the code clearer.